students/DiannaTingg/Session05/mailroom_three.py

```diff
@@ -0,0 +1,181 @@
+# Lesson 05 Assignment: Mailroom, Part 3
+
+# Import modules
+import datetime
+import os
+
+# Make a dictionary that holds donor names and a history of the amounts they donated
+donors = {"Jimmy Fallon": [500.00, 700.00], "Taylor Swift": [1000.00], "Dan White": [200.00, 300.00, 400.00],
+          "Trevor Noah": [750.00, 850.00], "Elon Musk": [3000.00], "Selena Gomez": [100.00, 150.00, 200.00]}
+
+
+# Display menu options
+def display_menu(menu_dict):
+    answer = ""
+
+    # Until the user selects Q for quit
+    while answer != "Q":
+        print("\nMAIN MENU"
+              "\nType T to Send a Thank You to a Single Donor"
+              "\nType R to Create a Report"
+              "\nType A to Send Thank You Letters to All Donors"
+              "\nType Q to Quit")
+
+        answer = input("\nPlease enter a command: ")
+
+        # Get first character of user input and capitalize it
+        try:
+            answer = answer[0].capitalize()
+
+        # If user enters an empty string
+        except IndexError:
+            pass
+
+        # Get menu selection from dictionary
+        try:
+            menu_dict.get(answer)()
+
+        # If user types an invalid letter or number
+        except TypeError:
```

Comment:
what would raise a TypeError here?

Reply:
ah - got it -- if get() defaults to None -- so you'd get a TypeError if it's not in the dict.

That works!